src/Controllers/AdminController.php

```
@@ -1467,45 +1478,50 @@ public function saveSettings(): void
             }
-            $apiKey = $saved['ai']['api_key'] ?? '';
+            $savedProvider = AiProviderRegistry::normalize((string) ($saved['ai']['provider'] ?? ''));
+            if ($savedProvider === $provider) {
+                $apiKey = (string) ($saved['ai']['api_key'] ?? '');
+            }
         }
 
-        // Only Google Gemini is supported as primary provider; others return a clear message
-        $supported = ['', 'google'];
-        if (!in_array($provider, $supported, true)) {
+        if (!AiProviderRegistry::isSupported($provider)) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\StratFlow\Services\AiProviderRegistry' in method 'testAi'. (undefined)

(StaticAccess)